rosbag2_cpp/src/rosbag2_cpp/info.cpp

```diff
@@ -157,12 +293,13 @@ std::vector<std::shared_ptr<rosbag2_service_info_t>> Info::read_service_info(
       all_service_info[topic_name]->response_count = count;
     }
 
+    // Convert all_service_info to output_service_info
     for (auto & [topic_name, service_info] : all_service_info) {
-      ret_service_info.emplace_back(std::move(service_info));
+      output_service_info.emplace_back(std::move(service_info));
     }
   }
 
-  return ret_service_info;
+  return;
```

Comment:
Suggested to remove this return statement. The reasoning is a Clang-Tidy warning "Redundant return statement at the end of a function with a void return type"
```suggestion
```